hogql_parser/string.cpp

```diff
@@ -0,0 +1,56 @@
+#include <boost/algorithm/string.hpp>
+
+#include "error.h"
+#include "string.h"
+
+using namespace std;
+
+string unquote_string(string text) {
+  size_t original_text_size = text.size();
+  if (original_text_size == 0) {
+    throw HogQLParsingException("Encountered an unexpected empty string input");
+  }
+  const char first_char = text.front();
+  const char last_char = text.back();
+  if (first_char == '\'' && last_char == '\'') {
+    text = text.substr(1, original_text_size - 2);
+    boost::replace_all(text, "''", "'");
+    boost::replace_all(text, "\\'", "'");
+  } else if (first_char == '"' && last_char == '"') {
+    text = text.substr(1, original_text_size - 2);
+    boost::replace_all(text, "\"\"", "\"");
+    boost::replace_all(text, "\\\"", "\"");
+  } else if (first_char == '`' && last_char == '`') {
+    text = text.substr(1, original_text_size - 2);
+    boost::replace_all(text, "``", "`");
+    boost::replace_all(text, "\\`", "`");
+  } else if (first_char == '{' && last_char == '}') {
+    text = text.substr(1, original_text_size - 2);
+    boost::replace_all(text, "{{", "{");
+    boost::replace_all(text, "\\{", "{");
+  } else {
+    throw HogQLSyntaxException("Invalid string literal, must start and end with the same quote type: " + text);
+  }
+
+  // Copied from clickhouse_driver/util/escape.py
+  boost::replace_all(text, "\\a", "\a");
+  boost::replace_all(text, "\\b", "\b");
+  boost::replace_all(text, "\\f", "\f");
+  boost::replace_all(text, "\\n", "\n");
+  boost::replace_all(text, "\\r", "\r");
+  boost::replace_all(text, "\\t", "\t");
+  boost::replace_all(text, "\\v", "\v");
+  boost::replace_all(text, "\\0", "\0");
```

Comment:
Should null also be ignored here and elsewhere for consistency?

Reply:
Aligned with Python